core/src/main/java/com/bumble/appyx/core/node/Node.kt

```diff
@@ -95,14 +99,39 @@ abstract class Node(
             LocalNode provides this,
             LocalLifecycleOwner provides this,
         ) {
-            DerivedSetup()
+            HandleBackPress()
             View(modifier)
         }
     }
 
-    /** Derived classes can declare functional (non-ui) Composable blocks before [View()] is invoked. */
     @Composable
-    protected open fun DerivedSetup() {
+    private fun HandleBackPress() {
+        val backPressHandlerPlugins = remember { plugins.filterIsInstance<BackPressHandler>() }
+        val dispatcher = LocalOnBackPressedDispatcherOwner.current?.onBackPressedDispatcher
+        DisposableEffect(dispatcher) {
+            backPressHandlerPlugins.forEach { plugin ->
+                if (!plugin.isCorrect()) {
+                    Appyx.reportException(
+                        IllegalStateException("Plugin $plugin has implementation for both BackPressHandler properties, implement only one")
+                    )
+                }
+                plugin.onBackPressedCallbackList.forEach { dispatcher?.addCallback(it) }
+            }
+            onDispose {
+                backPressHandlerPlugins.forEach { plugin ->
+                    plugin.onBackPressedCallbackList.forEach { it.remove() }
+                }
+            }
+        }
+    }
+
+    // BackPressHandler is correct when only one of its properties is implemented.
+    private fun BackPressHandler.isCorrect(): Boolean {
+        val listIsOverriddenOrPluginIgnored = onBackPressedCallback == null
+        val onlySingleCallbackIsOverridden = onBackPressedCallback != null &&
+                onBackPressedCallbackList.size == 1 &&
+                onBackPressedCallbackList[0] == onBackPressedCallback
+        return onlySingleCallbackIsOverridden || listIsOverriddenOrPluginIgnored
```

Comment:
should this be `return onlySingleCallbackIsOverridden != listIsOverriddenOrPluginIgnored`?

Otherwise if they are both true it will still be deemed correct